server/test/serverTest.js

```diff
@@ -171,6 +173,134 @@ describe('Chatbot server', () => {
+            let installations = await db.getInstallations()
+            await db.createButton(unit1UUID, installations[0].id, "1", unit1PhoneNumber, unit1SerialNumber)
+            await db.createButton(unit2UUID, installations[0].id, "2", unit2PhoneNumber, unit2SerialNumber)
+        });
+
+        afterEach(async function() {
+            await db.clearSessions()
+            await db.clearButtons()
+            await db.clearInstallations()
+            console.log('\n')
+        });
+
+        it('should return 400 to a request with no headers', async () => {
+            let response = await chai.request(server).post('/flic_button_press').send({})
+            expect(response).to.have.status(400)
+        });
+
+        it('should return 400 to a request with only button-serial-number', async () => {
+            let responseNoBatteryLEvel = await chai.request(server).post('/flic_button_press').set('button-serial-number', '100').send({})
+            expect(responseNoBatteryLEvel).to.have.status(400)
+        });
+
+        it('should return 400 to a request with only button-battery-level', async () => {
+            let responseNoSerialNumber = await chai.request(server).post('/flic_button_press').set('button-battery-level', '100').send({})
+            expect(responseNoSerialNumber).to.have.status(400)
+        });
+
+
+        it('should return 400 to a request with an unregistered button', async () => {
+            let response = await chai.request(server).post('/flic_button_press').set('button-serial-number', 'CCCC-C0C0C0').set('button-battery-level', '100').send()
+            expect(response).to.have.status(400)
+        });
+
+        it('should return 200 to a valid request', async () => {
+            let response = await chai.request(server).post('/flic_button_press').set('button-serial-number', unit1SerialNumber).set('button-battery-level', '100').send()
+            expect(response).to.have.status(200)
+        });
+
+        it('should be able to create a valid session state from valid request', async () => {
+            let response = await chai.request(server).post('/flic_button_press').set('button-serial-number', unit1SerialNumber).set('button-battery-level', '100').send()
+            expect(response).to.have.status(200)
+
+            let sessions = await db.getAllSessionsWithButtonId(unit1UUID)
+            expect(sessions.length).to.equal(1)
+            
+            let session = sessions[0]
+            expect(session).to.not.be.null
+            expect(session).to.have.property('buttonId')
+            expect(session).to.have.property('unit')
+            expect(session).to.have.property('state')
+            expect(session).to.have.property('numPresses')
+            expect(session.buttonId).to.deep.equal(unit1UUID)
+            expect(session.unit).to.deep.equal('1')
+            expect(session.numPresses).to.deep.equal(1)
```

Comment:
these last two lines (and maybe others) can just be a strict equality check, not deep equality